src/ebonite/runtime/server/base.py

```diff
@@ -50,3 +54,65 @@ def start(self, loader: InterfaceLoader):
         interface = loader.load()
         rlogger.info('Running server %s', self)
         return self.run(interface)
+
+
+class HTTPServerConfig(Config):
+    host = Param('host', default='0.0.0.0', parser=str)
+    port = Param('port', default='9000', parser=int)
+
+
+if Core.DEBUG:
+    HTTPServerConfig.log_params()
+
+
+class MalformedHTTPRequestException(Exception):
+    def __init__(self, message: str):
+        self._message = message
+
+    def code(self):
+        return 400
+
+    def response_body(self):
+        return {'ok': False, 'error': self._message}
+
+
+class BaseHTTPServer(Server):
+    """
+    HTTP-based Ebonite runtime server.
+
+    Interface definition is exposed for clients via HTTP GET call to `/interface.json`,
+    method calls - via HTTP POST calls to `/<name>`,
+    server health check - via HTTP GET call to `/health`.
+
+    Host to which server binds is configured via `EBONITE_HOST` environment variable:
+    default is `0.0.0.0` which means any local or remote, for rejecting remote connections use `localhost` instead.
+
+    Port to which server binds to is configured via `EBONITE_PORT` environment variable: default is 9000.
+    """
+
+    @staticmethod
+    def _deserialize_json(interface: Interface, method: str, request_json: dict):
```

Comment:
why private method?

Reply:
uhm, this method is protected as it is expected to be run by subclasses only, not by clients.

Reply:
but you call it not from subclasses in flask and aiohttp code

Reply:
Yes but it is called from methods which are essentially implementation details of corresponding servers. Ideally these methods should be moved to servers but I decided not to bother with this as it makes no real difference.